### PR TITLE
Fix broken API when the Admin API is disabled

### PR DESCRIPTION
**Pull request by maerteijn**
This will fix the issues in #193 (which actually means OscarAPI is broken when the Admin API is disabled)

Features:
- Separate `StockRecord` serializers and views for the Admin API and the regular API.
- Added a `DrillDownHyperlinkedRelatedField` to support urls with multiple lookup fields
- Run the complete testsuite with the Admin API enabled and disabled to make sure nothing breaks

### CODE REVIEW

File: oscarapi/views/product.py [hunk 1]
```diff
@@ -73,6 +77,20 @@ def get(
         return Response(ser.data)
 
 
+class ProductStockRecords(generics.ListAPIView):
+    serializer_class = ProductStockRecordSerializer
+    queryset = StockRecord.objects.all()
+
+    def get_queryset(self):
+        product_pk = self.kwargs.get("pk")
+        return super().get_queryset().filter(product__pk=product_pk)
```
Comment (specialunderwear) [inline oscarapi/views/product.py]: if you make this product_id=product_pk there is no join

File: oscarapi/serializers/product.py [hunk 2]
```diff
@@ -305,6 +307,17 @@ class Meta:
         fields = overridable("OSCARAPI_RECOMMENDED_PRODUCT_FIELDS", default=("url",))
 
 
+class ProductStockRecordSerializer(OscarModelSerializer):
+    url = DrillDownHyperlinkedIdentityField(
+        view_name="product-stockrecord-detail",
+        extra_url_kwargs={"product_pk": "product.id"},
```
Comment (specialunderwear) [inline oscarapi/serializers/product.py]: If this works with
```
 extra_url_kwargs={"product_pk": "product_id"}
```
that is 1 less query.

File: oscarapi/serializers/checkout.py [hunk 3]
```diff
@@ -134,6 +138,11 @@ class OrderLineSerializer(OscarHyperlinkedModelSerializer):
     price_excl_tax_excl_discounts = serializers.DecimalField(
         decimal_places=2, max_digits=12, source="line_price_before_discounts_excl_tax"
     )
+    stockrecord = DrillDownHyperlinkedRelatedField(
+        view_name="product-stockrecord-detail",
+        extra_url_kwargs={"product_pk": "product.id"},
```
Comment (specialunderwear) [inline oscarapi/serializers/checkout.py]: If this works with
```
 extra_url_kwargs={"product_pk": "product_id"}
```
that is 1 less query.

File: oscarapi/serializers/basket.py [hunk 4]
```diff
@@ -159,6 +159,12 @@ class BasketLineSerializer(OscarHyperlinkedModelSerializer):
         read_only=True, required=False, source="get_warning"
     )
 
+    stockrecord = DrillDownHyperlinkedRelatedField(
+        view_name="product-stockrecord-detail",
+        extra_url_kwargs={"product_pk": "product.id"},
```
Comment (specialunderwear) [inline oscarapi/serializers/basket.py]: If this works with
```
 extra_url_kwargs={"product_pk": "product_id"}
```
that is 1 less query.

File: oscarapi/serializers/admin/partner.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    OscarHyperlinkedModelSerializer,
+    UpdateListSerializer,
+)
+
+Product = get_model("catalogue", "Product")
+StockRecord = get_model("partner", "StockRecord")
+
+
+class AdminStockRecordSerializer(
+    DelayUniqueSerializerMixin, OscarHyperlinkedModelSerializer
```
Comment (specialunderwear) [inline oscarapi/serializers/admin/partner.py]: ❤️

Comment (maerteijn) [reply oscarapi/serializers/admin/partner.py]: 👍

**Comment by maerteijn** [comment]
Adjusted according to your comments

**Comment by SmileyJames** [comment]
Thanks you

**Pull request merged**